app/main/views/brief_responses.py

```diff
@@ -0,0 +1,117 @@
+from flask import jsonify, abort, request, current_app
+from sqlalchemy.exc import IntegrityError, DataError
+
+from dmapiclient.audit import AuditTypes
+
+from .. import main
+from ...models import db, Brief, BriefResponse, AuditEvent
+from ...utils import (
+    get_json_from_request, json_has_required_keys, get_int_or_400,
+    pagination_links, get_valid_page_or_1, url_for,
+    validate_and_return_updater_request,
+)
+
+from ...service_utils import validate_and_return_supplier
+
+
+@main.route('/brief-responses', methods=['POST'])
+def create_brief_response():
+    json_payload = get_json_from_request()
+    updater_json = validate_and_return_updater_request()
+
+    json_has_required_keys(json_payload, ['briefResponses'])
+    brief_response_json = json_payload['briefResponses']
+    json_has_required_keys(brief_response_json, ['briefId', 'supplierId'])
+
+    try:
+        brief = Brief.query.get(brief_response_json['briefId'])
+    except DataError:
+        brief = None
+
+    if brief is None:
+        abort(400, "Invalid brief ID '{}'".format(brief_response_json['briefId']))
+
+    if brief.status != 'live':
+        abort(400, "Brief must be live")
+
+    if brief.framework.status != 'live':
+        abort(400, "Brief framework must be live")
+
+    supplier = validate_and_return_supplier(brief_response_json)
+
+    # Check if brief response already exists from this supplier
+    if BriefResponse.query.filter(BriefResponse.supplier == supplier, BriefResponse.brief == brief).first():
+        abort(400, "Brief response already exists for supplier '{}'".format(supplier.supplier_id))
+
+    brief_response = BriefResponse(
+        data=brief_response_json,
```

Comment:
Should we strip the `briefId` and `supplierId` out of the `brief_response_json` before saving, as we have the linked `brief` and `supplier` objects as columns.  If we do strip them out here though then we'll need to preserve the submitted JSON for the audit event, otherwise it will be impossible to track anything from the auditing.